### ISSUE TITLE
Error: No available node types can fulfill resource request defaultdict

### ISSUE THREAD

**Issue by Kang9779** [GAIR-NLP/DeepResearcher#6]
环境配置：2*A100*40GB
NVIDIA-SMI 470.103.01   Driver Version: 470.103.01   CUDA Version: 11.8 

bash train_grpo.sh

/home/sankuai/conda/envs/deepresearcher/lib/python3.10/site-packages/torch/cuda/__init__.py:128: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11080). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
  return torch._C._cuda_getDeviceCount() > 0
2025-04-09 20:39:38,987 INFO worker.py:1660 -- Connecting to existing Ray cluster at address: 10.164.70.6:6379...
2025-04-09 20:39:38,998 INFO worker.py:1843 -- Connected to Ray cluster. View the dashboard at 127.0.0.1:8265 
(pid=31477) /home/sankuai/conda/envs/deepresearcher/lib/python3.10/site-packages/torch/cuda/__init__.py:128: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11080). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
(pid=31477)   return torch._C._cuda_getDeviceCount() > 0
(main_task pid=31477) {'actor_rollout_ref': {'actor': {'clip_ratio': 0.2,
(main_task pid=31477)                                  'entropy_coeff': 0.001,
(main_task pid=31477)                                  'fsdp_config': {'fsdp_size': -1,
(main_task pid=31477)                                                  'optimizer_offload': True,
(main_task pid=31477)                                                  'param_offload': True,
(main_task pid=31477)                                                  'wrap_policy': {'min_num_params': 0}},
(main_task pid=31477)                                  'grad_clip': 1.0,
(main_task pid=31477)                                  'kl_loss_coef': 0.001,
(main_task pid=31477)                                  'kl_loss_type': 'low_var_kl',
(main_task pid=31477)                                  'optim': {'lr': 1e-06,
(main_task pid=31477)                                            'lr_warmup_steps_ratio': 0.0,
(main_task pid=31477)                                            'min_lr_ratio': None,
(main_task pid=31477)                                            'total_training_steps': -1,
(main_task pid=31477)                                            'warmup_style': 'constant'},
(main_task pid=31477)                                  'ppo_epochs': 1,
(main_task pid=31477)                                  'ppo_max_token_len_per_gpu': 12288,
(main_task pid=31477)                                  'ppo_micro_batch_size': None,
(main_task pid=31477)                                  'ppo_micro_batch_size_per_gpu': 2,
(main_task pid=31477)                                  'ppo_mini_batch_size': 4096,
(main_task pid=31477)                                  'shuffle': False,
(main_task pid=31477)                                  'strategy': 'fsdp',
(main_task pid=31477)                                  'ulysses_sequence_parallel_size': 4,
(main_task pid=31477)                                  'use_dynamic_bsz': True,
(main_task pid=31477)                                  'use_kl_loss': True},
(main_task pid=31477)                        'hybrid_engine': True,
(main_task pid=31477)                        'model': {'enable_gradient_checkpointing': True,
(main_task pid=31477)                                  'external_lib': None,
(main_task pid=31477)                                  'override_config': {},
(main_task pid=31477)                                  'path': '/home/sankuai/dolphinfs_zhangkang20/ModelRepo/huggingface.co/Qwen/Qwen2.5-7B-Instruct',
(main_task pid=31477)                                  'use_remove_padding': True},
(main_task pid=31477)                        'ref': {'fsdp_config': {'param_offload': True,
(main_task pid=31477)                                                'wrap_policy': {'min_num_params': 0}},
(main_task pid=31477)                                'log_prob_max_token_len_per_gpu': 12288,
(main_task pid=31477)                                'log_prob_micro_batch_size': None,
(main_task pid=31477)                                'log_prob_micro_batch_size_per_gpu': 2,
(main_task pid=31477)                                'log_prob_use_dynamic_bsz': True,
(main_task pid=31477)                                'ulysses_sequence_parallel_size': 4},
(main_task pid=31477)                        'rollout': {'disable_log_stats': True,
(main_task pid=31477)                                    'do_sample': True,
(main_task pid=31477)                                    'dtype': 'bfloat16',
(main_task pid=31477)                                    'enable_chunked_prefill': True,
(main_task pid=31477)                                    'enforce_eager': True,
(main_task pid=31477)                                    'free_cache_engine': True,
(main_task pid=31477)                                    'gpu_memory_utilization': 0.6,
(main_task pid=31477)                                    'ignore_eos': False,
(main_task pid=31477)                                    'load_format': 'dummy_dtensor',
(main_task pid=31477)                                    'log_prob_max_token_len_per_gpu': 12288,
(main_task pid=31477)                                    'log_prob_micro_batch_size': None,
(main_task pid=31477)                                    'log_prob_micro_batch_size_per_gpu': 2,
(main_task pid=31477)                                    'log_prob_use_dynamic_bsz': True,
(main_task pid=31477)                                    'max_model_len': 32768,
(main_task pid=31477)                                    'max_num_batched_tokens': 32768,
(main_task pid=31477)                                    'max_num_seqs': 1024,
(main_task pid=31477)                                    'n': 1,
(main_task pid=31477)                                    'name': 'vllm',
(main_task pid=31477)                                    'prompt_length': 30767,
(main_task pid=31477)                                    'response_length': 2000,
(main_task pid=31477)                                    'temperature': 1.0,
(main_task pid=31477)                                    'tensor_model_parallel_size': 1,
(main_task pid=31477)                                    'top_k': -1,
(main_task pid=31477)                                    'top_p': 1,
(main_task pid=31477)                                    'use_fire_sampling': False}},
(main_task pid=31477)  'agent_grpo': {'n': 16},
(main_task pid=31477)  'algorithm': {'adv_estimator': 'grpo',
(main_task pid=31477)                'gamma': 1.0,
(main_task pid=31477)                'kl_ctrl': {'kl_coef': 0.001, 'type': 'fixed'},
(main_task pid=31477)                'kl_penalty': 'kl',
(main_task pid=31477)                'lam': 1.0},
(main_task pid=31477)  'critic': {'cliprange_value': 0.5,
(main_task pid=31477)             'forward_max_token_len_per_gpu': 32768,
(main_task pid=31477)             'forward_micro_batch_size': None,
(main_task pid=31477)             'forward_micro_batch_size_per_gpu': 2,
(main_task pid=31477)             'grad_clip': 1.0,
(main_task pid=31477)             'model': {'enable_gradient_checkpointing': True,
(main_task pid=31477)                       'external_lib': None,
(main_task pid=31477)                       'fsdp_config': {'fsdp_size': -1,
(main_task pid=31477)                                       'optimizer_offload': False,
(main_task pid=31477)                                       'param_offload': False,
(main_task pid=31477)                                       'wrap_policy': {'min_num_params': 0}},
(main_task pid=31477)                       'override_config': {},
(main_task pid=31477)                       'path': '/home/sankuai/dolphinfs_zhangkang20/ModelRepo/huggingface.co/Qwen/Qwen2.5-7B-Instruct',
(main_task pid=31477)                       'tokenizer_path': '/home/sankuai/dolphinfs_zhangkang20/ModelRepo/huggingface.co/Qwen/Qwen2.5-7B-Instruct',
(main_task pid=31477)                       'use_remove_padding': True},
(main_task pid=31477)             'optim': {'lr': 1e-05,
(main_task pid=31477)                       'lr_warmup_steps_ratio': 0.0,
(main_task pid=31477)                       'min_lr_ratio': None,
(main_task pid=31477)                       'total_training_steps': -1,
(main_task pid=31477)                       'warmup_style': 'constant'},
(main_task pid=31477)             'ppo_epochs': 1,
(main_task pid=31477)             'ppo_max_token_len_per_gpu': 32768,
(main_task pid=31477)             'ppo_micro_batch_size': None,
(main_task pid=31477)             'ppo_micro_batch_size_per_gpu': 2,
(main_task pid=31477)             'ppo_mini_batch_size': 4096,
(main_task pid=31477)             'shuffle': False,
(main_task pid=31477)             'strategy': 'fsdp',
(main_task pid=31477)             'ulysses_sequence_parallel_size': 4,
(main_task pid=31477)             'use_dynamic_bsz': True},
(main_task pid=31477)  'data': {'data_writing_file': './signal/data.json',
(main_task pid=31477)           'image_key': 'images',
(main_task pid=31477)           'max_model_len': 32768,
(main_task pid=31477)           'max_prompt_length': 30767,
(main_task pid=31477)           'max_response_length': 2000,
(main_task pid=31477)           'n_gpus_per_node': 8,
(main_task pid=31477)           'prompt_key': 'prompt',
(main_task pid=31477)           'query_signal': 1,
(main_task pid=31477)           'response_signal': 0,
(main_task pid=31477)           'return_raw_chat': False,
(main_task pid=31477)           'return_raw_input_ids': False,
(main_task pid=31477)           'shuffle': True,
(main_task pid=31477)           'signal_writing_file': './signal/signal.json',
(main_task pid=31477)           'tokenizer': None,
(main_task pid=31477)           'train_batch_size': 256,
(main_task pid=31477)           'train_files': './data/train.parquet',
(main_task pid=31477)           'val_batch_size': None,
(main_task pid=31477)           'val_files': './data/dev.parquet'},
(main_task pid=31477)  'do_search': True,
(main_task pid=31477)  'max_turns': 10,
(main_task pid=31477)  'reward_model': {'enable': False,
(main_task pid=31477)                   'forward_max_token_len_per_gpu': 32768,
(main_task pid=31477)                   'max_length': None,
(main_task pid=31477)                   'micro_batch_size': None,
(main_task pid=31477)                   'micro_batch_size_per_gpu': None,
(main_task pid=31477)                   'model': {'external_lib': None,
(main_task pid=31477)                             'fsdp_config': {'fsdp_size': -1,
(main_task pid=31477)                                             'min_num_params': 0,
(main_task pid=31477)                                             'param_offload': False},
(main_task pid=31477)                             'input_tokenizer': '/home/sankuai/dolphinfs_zhangkang20/ModelRepo/huggingface.co/Qwen/Qwen2.5-7B-Instruct',
(main_task pid=31477)                             'path': '~/models/FsfairX-LLaMA3-RM-v0.1',
(main_task pid=31477)                             'use_remove_padding': True},
(main_task pid=31477)                   'reward_manager': 'naive',
(main_task pid=31477)                   'strategy': 'fsdp',
(main_task pid=31477)                   'ulysses_sequence_parallel_size': 2,
(main_task pid=31477)                   'use_dynamic_bsz': True},
(main_task pid=31477)  'search_engine': 'online_search',
(main_task pid=31477)  'trainer': {'critic_warmup': 0,
(main_task pid=31477)              'default_hdfs_dir': None,
(main_task pid=31477)              'default_local_dir': './ckpts/project_name/experiment_name',
(main_task pid=31477)              'del_local_ckpt_after_load': False,
(main_task pid=31477)              'experiment_name': 'experiment_name',
(main_task pid=31477)              'logger': ['console', 'wandb'],
(main_task pid=31477)              'n_gpus_per_node': 8,
(main_task pid=31477)              'nnodes': 1,
(main_task pid=31477)              'project_name': 'project_name',
(main_task pid=31477)              'remove_previous_ckpt_in_save': False,
(main_task pid=31477)              'resume_from_path': False,
(main_task pid=31477)              'resume_mode': 'auto',
(main_task pid=31477)              'save_freq': 1,
(main_task pid=31477)              'test_freq': 1,
(main_task pid=31477)              'total_epochs': 1,
(main_task pid=31477)              'total_training_steps': None,
(main_task pid=31477)              'val_before_train': False,
(main_task pid=31477)              'val_generations_to_log_to_wandb': 0}}
(main_task pid=31477) WARNING 04-09 20:39:45 _custom_ops.py:19] Failed to import from vllm._C with ImportError('/home/sankuai/conda/envs/deepresearcher/lib/python3.10/site-packages/vllm/_C.abi3.so: undefined symbol: cuTensorMapEncodeTiled')
(main_task pid=31477) /home/sankuai/conda/envs/deepresearcher/lib/python3.10/site-packages/vllm/connections.py:8: RuntimeWarning: Failed to read commit hash:
(main_task pid=31477) No module named 'vllm._version'
(main_task pid=31477)   from vllm.version import __version__ as VLLM_VERSION
(main_task pid=31477) update signal file finish!
(main_task pid=31477) [validate_config] All configuration checks passed successfully!
(main_task pid=31477) original dataset len: 80000
(main_task pid=31477) filter dataset len: 80000
(main_task pid=31477) original dataset len: 875
(main_task pid=31477) filter dataset len: 875
(main_task pid=31477) Size of train dataloader: 312
(main_task pid=31477) Total training steps: 312
(autoscaler +30s) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
(autoscaler +30s) Error: No available node types can fulfill resource request defaultdict(<class 'float'>, {'CPU': 8.0, 'GPU': 8.0}). Add suitable node types to this cluster to resolve this issue.
(autoscaler +1m5s) Error: No available node types can fulfill resource request defaultdict(<class 'float'>, {'CPU': 8.0, 'GPU': 8.0}). Add suitable node types to this cluster to resolve this issue.

**Comment by Fu-Dayuan** [comment]
ulysses_sequence_parallel_size和n_gpus_per_node都要小于你拥有的卡数

**Comment by Kang9779** [comment]
> 而且我估计2A10040GB的有点少，可能训练跑不起来

有推荐的显存要求吗？

**Comment by Fu-Dayuan** [comment]
> > 而且我估计2A10040GB的有点少，可能训练跑不起来
> 
> 有推荐的显存要求吗？

长度卡32000的话4张A800 80G应该是下限（训练），推理应该一张80G的就可以
